internal/testhelper/testhelper.go

```diff
@@ -168,11 +167,10 @@ func SetupTestGRPCServer(tb testing.TB, logger golog.Logger) (*grpc.Server, int)
 	listener, err := net.Listen("tcp", ":0")
 	test.That(tb, err, test.ShouldBeNil)
 	grpcServer := grpc.NewServer()
-	goutils.PanicCapturingGo(func() {
-		if err := grpcServer.Serve(listener); err != nil {
-			logger.Errorw("error when starting up the grpc server", "error", err)
-		}
-	})
```

Comment:
Mind explaining what the issue was?

Reply:
This is the data race log:
```
2023-03-27T18:00:50.400Z	DEBUG	viam-cartographer/viam-cartographer.go:515	unchecked error	***"error": "rpc error: code = Canceled desc = grpc: the client connection is closing"***
PASS
==================
WARNING: DATA RACE
Read at 0x00c000a50383 by goroutine 231:
  testing.(*common).logDepth()
      /usr/lib/go-1.19/src/testing/testing.go:883 +0xc4
  testing.(*common).log()
      /usr/lib/go-1.19/src/testing/testing.go:876 +0xa4
  testing.(*common).Logf()
      /usr/lib/go-1.19/src/testing/testing.go:927 +0x6a
  testing.(*T).Logf()
      <autogenerated>:1 +0x75
  go.uber.org/zap/zaptest.testingWriter.Write()
      /home/testbot/go/pkg/mod/go.uber.org/zap@v1.24.0/zaptest/logger.go:130 +0x12c
  go.uber.org/zap/zaptest.(*testingWriter).Write()
      <autogenerated>:1 +0x7e
  go.uber.org/zap/zapcore.(*ioCore).Write()
      /home/testbot/go/pkg/mod/go.uber.org/zap@v1.24.0/zapcore/core.go:99 +0x199
  go.uber.org/zap/zapcore.(*CheckedEntry).Write()
      /home/testbot/go/pkg/mod/go.uber.org/zap@v1.24.0/zapcore/entry.go:255 +0x2ce
  go.uber.org/zap.(*SugaredLogger).log()
      /home/testbot/go/pkg/mod/go.uber.org/zap@v1.24.0/sugar.go:295 +0x13a
  go.uber.org/zap.(*SugaredLogger).Errorw()
      /home/testbot/go/pkg/mod/go.uber.org/zap@v1.24.0/sugar.go:222 +0x137
  github.com/viamrobotics/viam-cartographer/internal/testhelper.SetupTestGRPCServer.func1()
      /__w/viam-cartographer/viam-cartographer/viam-cartographer/internal/testhelper/testhelper.go:173 +0x74
  go.viam.com/utils.PanicCapturingGoWithCallback.func1()
      /home/testbot/go/pkg/mod/go.viam.com/utils@v0.1.17/runtime.go:164 +0x72

Previous write at 0x00c000a50383 by goroutine 1[70](https://github.com/viamrobotics/viam-cartographer/actions/runs/4535051606/jobs/7990012353#step:7:71):
  testing.tRunner.func1()
      /usr/lib/go-1.19/src/testing/testing.go:1433 +0x7e4
  runtime.deferreturn()
      /usr/lib/go-1.19/src/runtime/panic.go:476 +0x32
  testing.(*T).Run.func1()
      /usr/lib/go-1.19/src/testing/testing.go:1493 +0x47

Goroutine 231 (running) created at:
  go.viam.com/utils.PanicCapturingGoWithCallback()
      /home/testbot/go/pkg/mod/go.viam.com/utils@v0.1.17/runtime.go:151 +0xd6
  go.viam.com/utils.PanicCapturingGo()
      /home/testbot/go/pkg/mod/go.viam.com/utils@v0.1.17/runtime.go:142 +0x1e5
  github.com/viamrobotics/viam-cartographer/internal/testhelper.SetupTestGRPCServer()
      /__w/viam-cartographer/viam-cartographer/viam-cartographer/internal/testhelper/testhelper.go:1[71](https://github.com/viamrobotics/viam-cartographer/actions/runs/4535051606/jobs/7990012353#step:7:72) +0x26
  github.com/viamrobotics/viam-cartographer_test.TestSLAMProcess.func3()
      /__w/viam-cartographer/viam-cartographer/viam-cartographer/viam-cartographer_test.go:346 +0x71
  testing.tRunner()
      /usr/lib/go-1.19/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      /usr/lib/go-1.19/src/testing/testing.go:1493 +0x47

Goroutine 170 (finished) created at:
  testing.(*T).Run()
      /usr/lib/go-1.19/src/testing/testing.go:1493 +0x75d
  testing.runTests.func1()
      /usr/lib/go-1.19/src/testing/testing.go:1846 +0x99
  testing.tRunner()
      /usr/lib/go-1.19/src/testing/testing.go:1446 +0x216
  testing.runTests()
      /usr/lib/go-1.19/src/testing/testing.go:1844 +0x7ec
  testing.(*M).Run()
      /usr/lib/go-1.19/src/testing/testing.go:1[72](https://github.com/viamrobotics/viam-cartographer/actions/runs/4535051606/jobs/7990012353#step:7:73)6 +0xa84
  main.main()
      _testmain.go:57 +0x2e9
==================
goroutine 283 [running]:
runtime/debug.Stack()
	/usr/lib/go-1.19/src/runtime/debug/stack.go:24 +0x72
runtime/debug.PrintStack()
	/usr/lib/go-1.19/src/runtime/debug/stack.go:16 +0x25
go.viam.com/utils.PanicCapturingGoWithCallback.func1.1()
	/home/testbot/go/pkg/mod/go.viam.com/utils@v0.1.17/runtime.go:154 +0x65
panic(***0x1d3dea0, 0xc000a3c0a0***)
	/usr/lib/go-1.19/src/runtime/panic.go:890 +0x262
testing.(*common).logDepth(0xc000a50340, ***0xc000464140, 0x91***, 0x3)
	/usr/lib/go-1.19/src/testing/testing.go:894 +0x6d9
testing.(*common).log(...)
	/usr/lib/go-1.19/src/testing/testing.go:876
testing.(*common).Logf(0xc000a50340, ***0x1fbef86, 0x2***, ***0xc000a3c080, 0x1, 0x1***)
	/usr/lib/go-1.19/src/testing/testing.go:927 +0xa5
go.uber.org/zap/zaptest.testingWriter.Write(***0x7f55eabcd6e8?, 0xc000a50340?***, 0x1?***, ***0xc0004c2400, 0x92, 0x400***)
	/home/testbot/go/pkg/mod/go.uber.org/zap@v1.24.0/zaptest/logger.go:130 +0x12d
go.uber.org/zap/zapcore.(*ioCore).Write(0xc000a5d4d0, ***0x2, ***0xc1009454998770e9, 0x5b37a870, 0x34f3920***, ***0x0, 0x0***, ***0x1fff718, 0x26***, ***0x1, ...***, ...***, ...)
	/home/testbot/go/pkg/mod/go.uber.org/zap@v1.24.0/zapcore/core.go:99 +0x19a
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc00065e000, ***0xc000a76000, 0x1, 0x2***)
	/home/testbot/go/pkg/mod/go.uber.org/zap@v1.24.0/zapcore/entry.go:255 +0x2cf
go.uber.org/zap.(*SugaredLogger).log(0xc000014e08, 0x2, ***0x1fff718, 0x26***, ***0x0, 0x0, 0x0***, ***0xc000675f30, 0x2, 0x2***)
	/home/testbot/go/pkg/mod/go.uber.org/zap@v1.24.0/sugar.go:295 +0x13b
go.uber.org/zap.(*SugaredLogger).Errorw(...)
	/home/testbot/go/pkg/mod/go.uber.org/zap@v1.24.0/sugar.go:222
github.com/viamrobotics/viam-cartographer/internal/testhelper.SetupTestGRPCServer.func1()
	/__w/viam-cartographer/viam-cartographer/viam-cartographer/internal/testhelper/testhelper.go:1[73](https://github.com/viamrobotics/viam-cartographer/actions/runs/4535051606/jobs/7990012353#step:7:74) +0x138
go.viam.com/utils.PanicCapturingGoWithCallback.func1()
	/home/testbot/go/pkg/mod/go.viam.com/utils@v0.1.17/runtime.go:164 +0x73
created by go.viam.com/utils.PanicCapturingGoWithCallback
	/home/testbot/go/pkg/mod/go.viam.com/utils@v0.1.17/runtime.go:151 +0xd7
2023-03-27T18:00:50.430Z	ERROR	utils@v0.1.17/runtime.go:155	panic while running function	***"error": "Log in goroutine after TestSLAMProcess has completed: 2023-03-27T18:00:50.428Z\tERROR\ttesthelper/testhelper.go:173\terror when starting up the grpc server\t***\"error\": \"grpc: the server has been stopped\"***"***
Found 1 data race(s)
FAIL	github.com/viamrobotics/viam-cartographer	1.5[76](https://github.com/viamrobotics/viam-cartographer/actions/runs/4535051606/jobs/7990012353#step:7:77)s
ok  	github.com/viamrobotics/viam-cartographer/internal/dim-2d	0.108s
?   	github.com/viamrobotics/viam-cartographer/internal/testhelper	[no test files]
?   	github.com/viamrobotics/viam-cartographer/module	[no test files]
FAIL
make: *** [Makefile:[88](https://github.com/viamrobotics/viam-cartographer/actions/runs/4535051606/jobs/7990012353#step:7:89): test-go] Error 1
Error: Process completed with exit code 2.
```

Reply:
It appears as if the race condition was related to logging. Furthermore, the old implementation would not actually fail the test if the server ever returned an error so I figured we could kill 2 birds with 1 stone by failing the test if there is an error & removing the dependence on logging which may or may not be thread safe.